momepy/functional/_dimension.py

```diff
@@ -157,3 +157,40 @@ def perimeter_wall(
     ].values
 
     return results
+
+
+def weighted_character(values: Series, areas: Series, graph: Graph) -> Series:
```

Comment:
```suggestion
def weighted_character(y: Series, areas: Series, graph: Graph) -> Series:
```

Custom variable shall always be `y` and shall support both numpy array and Series.